static/src/javascripts/projects/commercial/modules/brazeBanner.js

```diff
@@ -154,14 +155,25 @@ const getMessageFromBraze = async (apiKey: string, brazeUuid: string): Promise<b
             return;
         }
 
-        appboy.subscribeToInAppMessage((message: InAppMessage) => {
+        let subscriptionId: ?string;
+
+        const callback = (message: InAppMessage) => {
             if (message.extras) {
                 messageConfig = message;
                 resolve(true);
             } else {
                 resolve(false);
             }
-        });
+
+            // Unsubscribe
```

Comment:
Do we want to unsubscribe after *every* message, or only messages we actually show?

If an incompatible Braze message lands (e.g. without extras) then we might still want to wait for a good one to arrive. This will become more relevant if other teams start adopting Braze as a messaging channel.

Not urgent for now, but certainly something to think about.

Reply:
This is a good point 👍  and applies to the DCR version too. It also makes me realise that the `else` case with the `resolve(false)` just above can probably be removed for similar reasons. If the callback fires, but we _can't_ show anything (at this point I think this would only happen if there was a misconfiguration in Braze) then we may as well keep listening in case the callback fires again with data we _can_ use to show a message. I think fine to wait and do all this in a subsequent PR, but it seems worth doing.